api/jobs/handlers.py

```diff
@@ -0,0 +1,105 @@
+"""
+API request handlers for the jobs module
+"""
+
+from ..dao.containerutil import create_filereference_from_dictionary, create_containerreference_from_dictionary
+
+from .. import base
+from .. import config
+
+from .jobs import Job
+from .queue import Queue
+
+log = config.log
+
+
+class JobsHandler(base.RequestHandler):
+
+    """Provide /jobs API routes."""
+
+    def get(self):
+        """
+        List all jobs.
+        """
+        if not self.superuser_request:
+            self.abort(403, 'Request requires superuser')
+
+        return list(config.db.jobs.find())
+
+    def add(self):
+        """
+        Add a job to the queue.
+        """
+
+        # TODO: Check each input container for R, check dest container for RW
+        if not self.superuser_request:
+            self.abort(403, 'Request requires superuser')
+
+        submit = self.request.json
+        gear_name = submit['gear']
+
+        # Translate maps to FileReferences
+        inputs = {}
+        for x in submit['inputs'].keys():
+            input_map = submit['inputs'][x]
+            inputs[x] = create_filereference_from_dictionary(input_map)
+
+        # Add job tags, attempt number, and/or previous job ID, if present
+        tags            = submit.get('tags', None)
+        attempt_n       = submit.get('attempt_n', 1)
+        previous_job_id = submit.get('previous_job_id', None)
+
+        # Add destination container, if present
+        destination = None
+        if submit.get('destination', None) is not None:
+            destination = create_containerreference_from_dictionary(submit['destination'])
+
+        job = Job(gear_name, inputs, destination=destination, tags=tags, attempt=attempt_n, previous_job_id=previous_job_id)
+        return job.insert()
+
+    def stats(self):
+        if not self.superuser_request:
+            self.abort(403, 'Request requires superuser')
+
+        return Queue.get_statistics()
+
+    def next(self):
+        if not self.superuser_request:
+            self.abort(403, 'Request requires superuser')
+
+        job = Queue.start_job()
+
+        if job is None:
+            self.abort(400, 'No jobs to process')
```

Comment:
@kofalt Looks like you're still throwing a 400 when there are no jobs to process, so the fix for #202 will still work as intended. 

Going to open a ticket specifically addressing the handling of exceptions in non-handler classes from #66 so we can finalize the way we would like to pass server and client error exceptions from non-handler classes.